### PR TITLE
Network: Remove http.enabled setting

### LINKED ISSUE #12792: Get rid of http.enabled
Do we really need this?

To me it seems like a bad setting as it stops simple interactions with the cluster and also destroys any ability to do monitoring (especially with Marvel).

### PR DESCRIPTION
**Pull request by rjernst**
This commit removes the `http.enabled` setting. While all real nodes (started with bin/elasticsearch) will always have an http binding, there are many tests that rely on the quickness of not actually needing to bind to 2 ports. For this case, the `MockHttpTransport.TestPlugin` provides a dummy http transport implementation which is used by default in ESIntegTestCase.

closes #12792

### CODE REVIEW

**Comment by rjernst** [comment]
@pcsanwald I tried adding this as the first entry in the new CHANGELOG. Please let me know if this is what you were thinking.

**Comment by pcsanwald** [comment]
@rjernst the changelog part LGTM

File: docs/reference/migration/migrate_7_0/settings.asciidoc [hunk 1]
```diff
@@ -25,3 +25,10 @@
   the system property `es.thread_pool.write.use_bulk_as_display_name` was
   available to keep the display output in APIs as `bulk` instead of `write`.
   These fallback settings and this system property have been removed.
+
+[[remove-http-enabled]]
+==== Http enabled setting removed
+
+The setting `http.enabled` previously allowed disabling binding to http, only allowing
+use of the transport client. This setting has been removed, as the transport client
+will be removed in the future, thus requiring http to always be enabled.
```
Comment (jasontedor) [inline docs/reference/migration/migrate_7_0/settings.asciidoc]: An incredibly minor nit, sorry: `http` -> `HTTP`.

File: docs/reference/migration/migrate_7_0/settings.asciidoc [hunk 2]
```diff
@@ -25,3 +25,10 @@
   the system property `es.thread_pool.write.use_bulk_as_display_name` was
   available to keep the display output in APIs as `bulk` instead of `write`.
   These fallback settings and this system property have been removed.
+
+[[remove-http-enabled]]
+==== Http enabled setting removed
+
+The setting `http.enabled` previously allowed disabling binding to http, only allowing
```
Comment (jasontedor) [inline docs/reference/migration/migrate_7_0/settings.asciidoc]: `http` -> `HTTP`.

**Review by jasontedor — APPROVED** [review]
LGTM, only a nit.

**Comment by rjernst** [comment]
@elasticmachine test this please

**Pull request merged**